proxy.js

```diff
@@ -0,0 +1,71 @@
+'use strict';
+
+const processFn = (fn, opts) => function () {
+	const that = this;
+	const P = opts.promiseModule;
+	const args = new Array(arguments.length);
+
+	for (let i = 0; i < arguments.length; i++) {
+		args[i] = arguments[i];
+	}
```

Comment:
Unfortunately, the version below breaks optimization:

```js
const processFn = (fn, opts) => function (...args) { …
```

---

And the following version don't work well too:

```js
const processFn = (fn, opts) => function (args) { …
```

`handler.get` have to be able to return `processFn(x, opts)` with the old function signature. We could fix the signature in `handler.get`, but that wouldn't be very sensible in my opinion.